### PR TITLE
Update docs to add workload identity functionality related in data-plane

### PR DESCRIPTION
**Pull request by grac3gao**
## Proposed Changes
related on #540 
related on #671
- Update Installing Pub/Sub Enabled Service Account doc to include "how to enable workload identity in data-plane"
- Update channel, sources example docs to include "how to leverage workload identity when it is enabled"
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by grac3gao** [comment]
Ready for review for the data-plane part. 

One question, not sure what is the proper way to update pullsubsciption and topic docs, as we didn't explicitly enable workload identity for pullsubscription and topics. We implicitly leverage WI by using sources to create pullsubscription/topic and put k8s serviceaccount as the serviceaccount for publisher and receive-adaptor pod. 

I was wondering if I can just say topic and pullsubscription not support WI, or I should update all those command for creating gSA, kSA, iam-policy-binding in topic and pullsubscription doc?

File: docs/install/pubsub-service-account.md [hunk 1]
```diff
@@ -47,25 +47,40 @@ Service Account.
...
+        It is the recommended way to access Google Cloud services from within GKE due to its improved security properties and 
+        manageability. More information about [workload identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity). 
+        
+        1. Enable workload identity. Skip this step if it is already enabled in control plane.
+            ```shell
+           gcloud beta container clusters update ${CLUSTER_NAME} \
+            --identity-namespace=${PROJECT_ID}.svc.id.goog
+           ``` 
+        1. Update `spec.serviceAccount` with [Google Cloud Service Account](https://console.cloud.google.com/iam-admin/serviceaccounts/project)
+         when creating resources. Check docs to see [example](https://github.com/google/knative-gcp/tree/master/docs/examples) for each resource.
```
Comment (nachocano) [inline docs/install/pubsub-service-account.md]: [examples]

File: docs/install/pubsub-service-account.md [hunk 2]
```diff
@@ -47,25 +47,40 @@ Service Account.
           --member=serviceAccount:cre-pubsub@$PROJECT_ID.iam.gserviceaccount.com \
           --role roles/pubsub.editor
         ```
-
-    1.  Download a new JSON private key for that Service Account. **Be sure not
+1. Access Google Cloud service. There are two methods provided to access
```
Comment (nachocano) [inline docs/install/pubsub-service-account.md]: Configure the authentication mechanism used for accessing the Google Cloud services. Currently, we support two methods:

File: docs/install/pubsub-service-account.md [hunk 3]
```diff
@@ -47,25 +47,40 @@ Service Account.
           --member=serviceAccount:cre-pubsub@$PROJECT_ID.iam.gserviceaccount.com \
           --role roles/pubsub.editor
         ```
-
-    1.  Download a new JSON private key for that Service Account. **Be sure not
+1. Access Google Cloud service. There are two methods provided to access 
+    Google Cloud service with [Google Cloud Service Account](https://console.cloud.google.com/iam-admin/serviceaccounts/project),
+    you can either:
+    1. Use workload identity.
```
Comment (nachocano) [inline docs/install/pubsub-service-account.md]: Workload Identity

File: docs/install/pubsub-service-account.md [hunk 4]
```diff
@@ -47,25 +47,40 @@ Service Account.
...
+     
+        It is the recommended way to access Google Cloud services from within GKE due to its improved security properties and 
+        manageability. More information about [workload identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
```
Comment (nachocano) [inline docs/install/pubsub-service-account.md]: `For more information about Workload Identity see [here](....)`

File: docs/install/pubsub-service-account.md [hunk 5]
```diff
@@ -47,25 +47,40 @@ Service Account.
...
-    1.  Download a new JSON private key for that Service Account. **Be sure not
+1. Access Google Cloud service. There are two methods provided to access 
+    Google Cloud service with [Google Cloud Service Account](https://console.cloud.google.com/iam-admin/serviceaccounts/project),
+    you can either:
+    1. Use workload identity.
+     
+        It is the recommended way to access Google Cloud services from within GKE due to its improved security properties and 
+        manageability. More information about [workload identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity). 
+        
+        1. Enable workload identity. Skip this step if it is already enabled in control plane.
```
Comment (nachocano) [inline docs/install/pubsub-service-account.md]: everywhere you use `workload identity` in docs, change it to `Workload Identity`

Comment (nachocano) [reply docs/install/pubsub-service-account.md]: `Skip this step if you already enable it in the [control plane setup](....)`. and put the link to the control plane installation setup

File: docs/install/pubsub-service-account.md [hunk 6]
```diff
@@ -47,25 +47,40 @@ Service Account.
...
+            ```shell
+           gcloud beta container clusters update ${CLUSTER_NAME} \
+            --identity-namespace=${PROJECT_ID}.svc.id.goog
+           ``` 
+        1. Update `spec.serviceAccount` with [Google Cloud Service Account](https://console.cloud.google.com/iam-admin/serviceaccounts/project)
```
Comment (nachocano) [inline docs/install/pubsub-service-account.md]: with a

**Comment by nachocano** [comment]
> Ready for review for the data-plane part.
> 
> One question, not sure what is the proper way to update pullsubsciption and topic docs, as we didn't explicitly enable workload identity for pullsubscription and topics. We implicitly leverage WI by using sources to create pullsubscription/topic and put k8s serviceaccount as the serviceaccount for publisher and receive-adaptor pod.
> 
> I was wondering if I can just say topic and pullsubscription not support WI, or I should update all those command for creating gSA, kSA, iam-policy-binding in topic and pullsubscription doc?

We added in the topic.spec and ps.spec the `serviceAccount` field, right? If so, then I think we should also add the documentation in topic and ps... Folks might just want to use those lower-level primitives instead of sources, and they could also leverage workload identity... Similar thing with channel.... 
WDYT?

**Comment by grac3gao** [comment]
Oh, I see @nachocano. You are right. I messed up the configuration of topic and receiver adaptor...It shouldn't be a problem, I'll add it. Thank you!

**Comment by nachocano** [comment]
/lgtm
/approve

Thanks @grac3gao !!! Looks great
I think the only docs missing are the control plane ones, but I think we will have to generate two different artifacts in our release. Let's talk tmrw more

**Pull request merged**